src/App.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import './css/App.css';
+import Header from './Header';
+import Timer from './Timer';
+import Image from './Image';
+import {TimerConstants, MAX_SESSION_TIME, MIN_SESSION_TIME, MAX_BREAK_TIME, MIN_BREAK_TIME} from './Constants';
+
+class App extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            breakLength:TimerConstants.breakLength,
+            sessionLength:TimerConstants.sessionLength
+        }
+        this.increaseSessionLength = this.increaseSessionLength.bind(this);
+        this.decreaseSessionLength = this.decreaseSessionLength.bind(this);
+        this.increaseBreakLength = this.increaseBreakLength.bind(this);
+        this.decreaseBreakLength = this.decreaseBreakLength.bind(this);
+    }
+
+    reset = () => {
+        this.setState({
+            breakLength:TimerConstants.breakLength,
+            sessionLength:TimerConstants.sessionLength
+        });
+    }
+
+    increaseSessionLength = () => {
+        this.setState({sessionLength: (this.state.sessionLength < MAX_SESSION_TIME )?this.state.sessionLength+1:this.state.sessionLength});
```

Comment:
this.state.sessionLength < MAX_SESSION_TIME )?this.state.sessionLength+1:this.state.sessionLength yeh decision bahar hi le lete before setting state.